h/static/scripts/directives/via-dialog.coffee

```diff
@@ -0,0 +1,34 @@
+###*
+# @ngdoc directive
+# @name viaLink
+# @restrict A
+# @description The dialog that generates a via link to the page h is currently
+# loaded on.
+###
+
+viaLinkDialog = ['$timeout', ($timeout) ->
+    link: (scope, elem, attrs, ctrl) ->
+        ## Watch vialinkvisble: when it changes to true, focus input and selection.
+        scope.$watch (-> scope.viaLinkVisible), (visble) ->
+            if visble
+                $timeout (-> elem.find('#via').focus().select()), 0, false
+                
+    controller: ($scope, $element, $attrs) ->
+    	$scope.viaLinkVisible = false
+    	$scope.viaPageLink = 'https://via.hypothes.is/h/' + document.referrer
```

Comment:
IMO the correct thing here is to find a reference in specifications to the referrer attribute as it pertains to iframes to determine under what conditions, if any, a user agent SHOULD or SHOULD NOT make the URL of a parent frame available to a child frame. If anything, I would expect it to hinge on SSL/non-SSL or potentially only when there's a mix of the two in play. Finally, there are extensions and things that hide the referrer as a privacy feature, though who knows whether they would apply to iframes.

On the other hand, using the document plugin you could get the canonical href which would be a better choice that the referrer anyway.